src/term-spec.js

```diff
@@ -32,7 +32,7 @@ declare export class NamedObjectProperty extends ObjectProperty {
   name: PropertyName;
 }
 declare export class MethodDefinition extends NamedObjectProperty {
-  body: FunctionBody;
+  body?: FunctionBody; // undefined only on abstract class methods
```

Comment:
I think it's better to split it out into `MethodDefinition` with non-optional `body` and `AbstractMethodDefinition` with no `body`.